PROTO_tests/tests/array_view_test.py

```diff
@@ -0,0 +1,203 @@
+        array_ind = [pd_array[i, j] for (i, j) in product(range(3), repeat=2)]
+        assert nd_ind == reshape_ind
+        assert nd_ind == array_ind
+
+        with pytest.raises(IndexError):
+            # index out bounds (>= dimension)
+            # index 3 is out of bounds for axis 0 with size 3
+            pd_reshape[3, 1]
+        with pytest.raises(IndexError):
+            # index -4 is out of bounds for axis 1 with size 3
+            pd_reshape[2, -4]
+        with pytest.raises(IndexError):
+            # too many indicies for array: array is 2-dimensional, but 3 were indexed
+            pd_reshape[0, 1, 1]
+        with pytest.raises(ValueError):
+
+            # cannot reshape array of size 9 into shape (4,3)
+            ak.arange(9).reshape(4, 3)
+
+    @pytest.mark.parametrize("size", SIZE)
+    def test_int_list_indexing(self, size):
+        N = len(SHAPE[size])
+        iav = ak.arange(size).reshape(SHAPE[size])
+        uav = ak.arange(size, dtype=np.uint64).reshape(SHAPE[size])
+
+        iind = ak.ones(N, dtype=ak.uint64)
+        uind = ak.cast(iind, np.uint64)
+        assert np.array_equal(iav[iind], iav[uind])
+        assert np.array_equal(uav[iind], uav[uind])
+
+    @pytest.mark.parametrize("size", SIZE)
+    def test_set_index(self, size):
+        inav = np.arange(size).reshape(SHAPE[size])
+        unav = np.arange(size, dtype=np.uint64).reshape(SHAPE[size])
+        iav = ak.arange(size).reshape(SHAPE[size])
+        uav = ak.arange(size, dtype=ak.uint64).reshape(SHAPE[size])
+
+        nind = (1,) * inav.ndim
+        iind = ak.array(nind)
+        uind = ak.cast(iind, ak.uint64)
+
+        inav[nind] = -9999
+        unav[nind] = 2**64 - 9999
+        iav[uind] = -9999
+        uav[iind] = -9999
+        assert np.array_equal(iav[uind], inav[nind])
+        assert np.array_equal(iav[iind], iav[uind])
+        assert np.array_equal(uav[uind], unav[nind])
+        assert np.array_equal(uav[iind], uav[uind])
+
+    @pytest.mark.parametrize("size", SIZE)
+    def test_get_bool_pdarray(self, size):
+        N = len(SHAPE[size])
+
+        for i in range(0, N+1):
+            n = np.arange(size).reshape(SHAPE[size])
+            a = ak.arange(size).reshape(SHAPE[size])
+            truth = [True] * N
+            if i > 0:
+                truth[i - 1] = False
+            truth = tuple(truth)
+            n_bool_list = n[truth].tolist()
+            a_bool_list = a[truth].to_list()
+
+            assert n_bool_list == a_bool_list
+
+    @pytest.mark.parametrize("size", SIZE)
+    def test_set_bool_pdarray(self, size):
+        N = len(SHAPE[size])
+
+        for i in range(0, N+1):
+            n = np.arange(size).reshape(SHAPE[size])
+            a = ak.arange(size).reshape(SHAPE[size])
+            truth = [True] * N
+            if i > 0:
+                truth[i-1] = False
+            truth = tuple(truth)
+            n[truth] = 7
+            a[truth] = 7
+            assert n.tolist() == a.to_list()
```

Comment:
same suggestions for moving `n` and `a` out of loop and constructing truth using a tuple comprehension

Reply:
Updated `truth` construction, but keeping `n` and `a` inside the loop to regenerate between each test. Also added a random value to test each time